### ISSUE TITLE
Hide stops at high zoom levels

### ISSUE THREAD

**Issue by bbodenmiller** [OneBusAway/onebusaway-android#367]
Sometimes I need to pan around the map and when I zoom out it becomes very hard to do so because the map is constantly trying to redraw the stops/figure out which stops to show. Perhaps at certain zoom levels stops should be hidden.

**Comment by barbeau** [comment]
Agreed, this can get awkward, especially on slow wireless connections.

IMHO we can address this one of two ways:

1. Switch to "Nearby routes" mode when zoomed out (#352)
2. Cluster stops - there is an an [android-map-utils library](http://googlemaps.github.io/android-maps-utils/) that can help with this, but it would require a major re-write of how stops are rendered in OBA Android.

Solutions changing how stops are displayed when zoomed out are also constrained by the current limit of the stop-for-location API which only returns a max of 100 stops, as discussed in https://github.com/OneBusAway/onebusaway-application-modules/issues/153.  The set of stops returned by the API can differ on each API call, if more than 100 stops are contained in the bounding box.

**Comment by bbodenmiller** [comment]
I think a perfectly acceptable solution would be to just show no stops after a certain zoom level and just have a notice to zoom in to see stops. This is what OBA iOS does and I don't really see what value showing stops at low zoom levels provides.

**Comment by barbeau** [comment]
>I think a perfectly acceptable solution would be to just show no stops after a certain zoom level and just have a notice to zoom in to see stops. This is what OBA iOS does and I don't really see what value showing stops at low zoom levels provides.

My issue with this is that we've heard complaints from iOS users about not showing stops when zoomed out, but personally I've never heard complaints from Android users about showing stops when zoomed out.  Showing stops (or something) when zoomed out does give you a sense of area of coverage of OBA/transit system.

**Comment by barbeau** [comment]
Related to this, a user said the route line directions are hard to see a further zoomed out levels.  Right now the stops are still large, and end up obscuring the line itself.

If the line was visible, Google Maps API v2 now allows you to add custom line caps, which can be arrows - see https://stackoverflow.com/a/45494045/937715.

This would give us arrows at the end of lines, but I think we would need occasional flow arrows in the middle of the line as well - maybe that could be accomplished by splitting the route into a few polylines?

Related to https://github.com/OneBusAway/onebusaway-android/issues/444, simplifying route shapes before adding them to the map (to improve performance).
